packages/ra-core/src/controller/ListController.js

```diff
@@ -69,7 +69,18 @@ import removeKey from '../util/removeKey';
  *     );
  */
 export class ListController extends Component {
-    state = {};
+    constructor(props) {
+        super(props);
+
+        const enabledSources = {};
+        props.initiallyEnabledSources.forEach(
```

Comment:
Good place for reduce:
```js
const enabledSources = props.initiallyEnabledSources.reduce((sources, src) => ({ ...sources, [src]: true }), {})
```
(or some multi line version). You could also do
```js
const enabledSources = _.fromPairs(props.initiallyEnabledSources.map((src) => [src, true]))
```

Reply:
I like the reduce here and added it. @tomknig just update it in a follow up PR if you prefer something different.